src/datreant/tests/test_cli.py

```diff
@@ -0,0 +1,162 @@
+                                'spiky': True})
+    t6 = dtr.Treant('hausmann')
+    t7 = dtr.Treant('hausmann/head',
+                    tags=['hausmann'],
+                    categories={'material': 'wood',
+                                'spiky': False})
+
+
+def test_show(readymade_treant):
+    ret = subprocess.run('dtr show duchamp/fountain', shell=True,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE,
+                         check=True)
+
+    output = ret.stdout.decode()
+    assert 'art?' in output
+    assert 'colour' in output
+    assert 'white' in output
+
+
+def test_discover(readymades):
+    ret = subprocess.run('dtr discover', shell=True,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE,
+                         check=True,
+    )
+
+    output = ret.stdout.decode()
+
+    print(output)
+    items = [val for val in output.split('\n')
+             if val]  # ignore blank lines
+    assert len(items) == 7
+
+
+def test_get(readymades):
+    ret = subprocess.run('dtr get -c material:wood', shell=True,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE,
+                         check=True,
+    )
+
+    # should return 2 results, 'duchamp/wheel' and 'hausmann/head'
+    output = ret.stdout.decode()
+    print('#' + output + '#')
+    assert len(output.split('\n')) == 2
+    assert 'wheel' in output
+    assert 'head' in output
+
+
+def test_tags(readymade_treant):
+    ret = subprocess.run('dtr tags duchamp/fountain', shell=True,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE,
+                         check=True,
+    )
+
+    output = ret.stdout.decode()
+
+    assert 'art?' in output
+    assert 'duchamp' in output
+
+
+def test_categories(readymade_treant):
+    ret = subprocess.run('dtr categories duchamp/fountain', shell=True,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE,
+                         check=True,
+    )
+
+    output = ret.stdout.decode()
+
+    assert 'colour : white' in output
+
+
+def test_add_tag(readymade_treant):
+    ret = subprocess.run('dtr add duchamp/fountain -t dada',
+                         shell=True,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE,
+                         check=True,
+    )
+
+    t = dtr.Treant('duchamp/fountain')
+
+    assert 'dada' in t.tags
+
+
+def test_add_category(readymade_treant):
+    ret = subprocess.run('dtr add duchamp/fountain -c year:1917',
+                         shell=True,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE,
+                         check=True,
+    )
+
+    t = dtr.Treant('duchamp/fountain')
+
+    assert t.categories['year'] == 1917
```

Comment:
this fails because the value ends up as a string not an integer.  From a python shell it sets/recalls it as an integer, and seems the preferred behaviour?